proto/tendermint/types/types.proto

```diff
@@ -91,6 +91,7 @@ message Data {
   IntermediateStateRoots intermediate_state_roots = 2 [(gogoproto.nullable) = false];
   EvidenceList           evidence                 = 3 [(gogoproto.nullable) = false];
   Messages               messages                 = 4 [(gogoproto.nullable) = false];
+  uint64                 original_square_size     = 5;
```

Comment:
Since we don't store the data in the format that we commit to or the square size for each block, we cash the original square size in the data after we generate it. This is then saved and loaded when loading the block to serve a TxProof request.

We can change this in the future by saving adding things to `BlockMeta` or something similar.